include/api_manager/request.h

```diff
@@ -71,6 +71,9 @@ class Request {
   // Sets auth token to the request object. Caller of RequestHandler::Check
   // need to use it compose error message if authentication fails.
   virtual void SetAuthToken(const std::string &auth_token) = 0;
+  // Returns auth token. Used to compose authz_cache key in
+  // check_security_rules.cc
+  virtual std::string GetAuthToken() = 0;
```

Comment:
The old way to do this was exactly the same as the one you mentioned here: Check_auth set auth_token to request_context, and firebase gets it. But Limin said it would be better to get auth token directly from request.h. But there is no GetAuthToken function in request.h. So I decided to add one.

I agree that a dependency exists (check_auth.cc needs to set auth token before check_security_rules.cc can use it via Get AuthToken). But this will be okay, as authentication always goes before authorization. If authentication fails, ESP won't proceed to authorization.

In include/api_manager/request.h, I documented the usage of the GetAuthToken function.